test/test_mahalanobis_mixin.py

```diff
@@ -750,3 +726,35 @@ def test_deterministic_initialization(estimator, build_dataset):
   model2 = model2.fit(*remove_y(model, input_data, labels))
   np.testing.assert_allclose(model1.get_mahalanobis_matrix(),
                              model2.get_mahalanobis_matrix())
+
+
+@pytest.mark.parametrize('with_preprocessor', [True, False])
+@pytest.mark.parametrize('estimator, build_dataset', pairs_learners_m,
+                         ids=ids_pairs_learners_m)
+def test_raise_not_fitted_error_if_not_fitted(estimator, build_dataset,
+                                              with_preprocessor):
+  """Test that a NotFittedError is raised if someone tries to use
+  pair_score, pair_distance, score_pairs, get_metric, transform or
+  get_mahalanobis_matrix on input data and the metric learner
+  has not been fitted."""
+  input_data, _, preprocessor, _ = build_dataset(with_preprocessor)
+  estimator = clone(estimator)
+  estimator.set_params(preprocessor=preprocessor)
+  set_random_state(estimator)
+  with pytest.raises(NotFittedError):  # TODO: Remove in 0.8.0
+    msg = ("score_pairs will be deprecated in release 0.7.0. "
+           "Use pair_score to compute similarity scores, or "
+           "pair_distances to compute distances.")
+    with pytest.warns(FutureWarning) as raised_warning:
+      estimator.score_pairs(input_data)
+    assert any([str(warning.message) == msg for warning in raised_warning])
+  with pytest.raises(NotFittedError):
+    estimator.pair_score(input_data)
+  with pytest.raises(NotFittedError):
+    estimator.pair_distance(input_data)
+  with pytest.raises(NotFittedError):
+    estimator.get_metric()
+  with pytest.raises(NotFittedError):
+    estimator.get_mahalanobis_matrix()
+  with pytest.raises(NotFittedError):
+    estimator.transform(input_data)
```

Comment:
Some of these methods are present for all metric learners (`score_pairs`,  `pair_score`, `get_metric`) so they shoud be tested directly in `test_base_metric.py` instead of here

Additionally, there should be a similar test in `test_bilinear_mixin.py` for methods specific to bilinear similarities (such as `get_bilinear_matrix`)

Reply:
Should I build one generic test in `test_base_metric.py` for all kinds of leraners, at the cost of using `isinstance()` (as in `test_deprecated_score_pairs_same_result`),

or

Should I build one test for common functions in `test_base_metric.py`, and replicate the same test for specific functions in mahalanobis, bilinear and classifiers test files?